### PR TITLE
Fix for Issue #308: MockResolver: support "example" in definitions

### LINKED ISSUE #308: MockResolver: support "example" in definitions
Follow up to #292: we should also support returning example responses which are only defined indirectly via `example` fields in type definitions. Our current implementation only honors the `examples` property on the Response object in the Swagger spec.

### PR DESCRIPTION
**Pull request by gb-dsigmeth**
Fixes #308 .

Adding support for "example" in definitions to be returned from the mock server instead of "No example response was defined." for more robust mocking.

Changes proposed in this pull request:

 - If response example is not provided, use "example" defined in response type schema
 - If no response example or schema example is provided return error message as before

### CODE REVIEW

**Comment by hjacobs** [comment]
Thanks for the PR! :smile: 

It does not completely fix #308, as examples might be defined only for individual object properties, but not complete schema. Would you be willing to work towards full support for `example` properties in nested objects? (it's not trivial)

File: connexion/mock.py [hunk 1]
```diff
@@ -64,4 +64,14 @@ def mock_operation(self, operation, *args, **kwargs):
         if examples:
             return list(examples.values())[0], status_code
         else:
-            return 'No example response was defined.', status_code
+            # No response example, check for schema example
+            response_schema = response_definition.get('schema', {})
+            definitions = response_schema.get('definitions', {})
+            ref = response_schema.get('$ref', '')
```
Comment (hjacobs) [inline connexion/mock.py]: this would not work if the schema is defined inline

**Comment by gb-dsigmeth** [comment]
#409 more specifically addresses the concept of composing an example response from individual property examples, in contrast to #308 which refers only to full-schema examples. I am in impending need of the full-schema example functionality and found it's absence quite unexpected.

Conversely #409 is more of a nice-to-have feature that I never expected, and seems more limited in it's applicability. Whether I, or someone else, adds that functionality, it can just drop in after that final else, it can come in as it's own PR which directly references the #409 issue it solves.

Long story short: I consider fixing #308 valuable enough on it's own without necessarily solving #409 also.

**Comment by hjacobs** [comment]
:+1:
